compiler/catala_utils/message.ml

```diff
@@ -164,28 +164,117 @@ module Content = struct
+      | MainMessage msg ->
+        Format.fprintf ppf "@,@[<v 2>@,@[<hov>%t@]@]" msg;
+        if islast then Format.pp_print_cut ppf ();
+        true
+      | Position pos -> (
+        Format.pp_print_cut ppf ();
+        Option.iter
+          (fun msg -> Format.fprintf ppf "@[<v 1>@,@[<hov 2>%t@]@]" msg)
+          pos.pos_message;
+        Format.pp_print_break ppf 0 (-1);
+        let pr_head, pr_context, pr_legal = Pos.format_loc_text_parts pos.pos in
+        Format.pp_open_vbox ppf 2;
+        Format.fprintf ppf "@{<blue>@<1>%s@}%t" "├" pr_head;
+        pr_context ppf;
+        Format.pp_close_box ppf ();
+        match pr_legal with
+        | None -> true
+        | Some pr_legal ->
+          Format.pp_print_break ppf 0 (-1);
+          if islast then (
+            restore_ppf ();
+            Format.pp_open_vbox ppf 3;
+            Format.fprintf ppf "@{<blue>@<3>%s@}%t" "└─ " pr_legal)
+          else (
+            Format.pp_open_vbox ppf 3;
+            Format.fprintf ppf "@{<blue>@<3>%s@}%t" "├─ " pr_legal);
+          Format.pp_close_box ppf ();
+          not islast)
+      | Outcome msg ->
+        Format.fprintf ppf "@,@[<hov 2>%t@]" msg;
+        true
+      | Suggestion suggestions_list ->
+        Format.fprintf ppf "@,@[<v 1>@,@[<hov 2>%a@]@]" Suggestions.format
+          suggestions_list;
+        true
+    in
+    let rec print_lines ppf = function
+      | [elt] ->
+        let finalise = print_elt ppf ~islast:true elt in
+        Format.pp_close_box ppf ();
+        if finalise then Format.fprintf ppf "@,@{<blue>@<2>%s@}" "└─"
+      | elt :: r ->
+        let _ = print_elt ppf elt in
+        print_lines ppf r
+      | [] ->
+        Format.pp_close_box ppf ();
+        Format.pp_print_cut ppf ()
+    in
+    print_lines ppf content;
+    Format.pp_close_box ppf ();
+    restore_ppf ();
+    Format.pp_print_newline ppf ()
+
   let emit (content : t) (target : level) : unit =
     match Global.options.message_format with
-    | Global.Human ->
+    | Global.Human -> (
       let ppf = get_ppf target in
-      Format.pp_open_vbox ppf 0;
-      Format.pp_print_list
-        ~pp_sep:(fun ppf () -> Format.fprintf ppf "@,@,")
-        (fun ppf -> function
-          | Position pos ->
-            Option.iter
-              (fun msg -> Format.fprintf ppf "@[<hov>%t@]@," msg)
-              pos.pos_message;
-            Pos.format_loc_text ppf pos.pos
-          | MainMessage msg ->
-            Format.fprintf ppf "@[<hov 2>%t %t@]" (pp_marker target) msg
-          | Outcome msg ->
-            Format.fprintf ppf "@[<hov>%t@ %t@]" (pp_marker target) msg
-          | Suggestion suggestions_list ->
-            Suggestions.format ppf suggestions_list)
-        ppf content;
-      Format.pp_close_box ppf ();
-      Format.pp_print_newline ppf ()
+      match target with
+      | Debug | Log -> basic_msg ppf target content
+      | Result | Warning | Error -> fancy_msg ppf target content)
```

Comment:
Interesting, what's the rationale for printing debug and log messages in a degraded format?

Reply:
Better answered with a screenshot:
![image](https://github.com/CatalaLang/catala/assets/807966/4a343f9c-2aaf-435f-8b1a-290e9c7a1d65)

---
If we were to use the "fancy" printer:
![image](https://github.com/CatalaLang/catala/assets/807966/c3c768e0-3bfe-426d-b0be-1cc0d52f0db7)

Reply:
I see, that's a reasonable decision, thanks!